### PR TITLE
feat: Added ORC and PARQUET support among source tests

### LINKED ISSUE #1252: Support parquet and ORC files for Filesystem validations
Pandas supports [read_parquet()](https://pandas.pydata.org/pandas-docs/version/1.3/reference/api/pandas.read_parquet.html) and [read_orc()](https://pandas.pydata.org/docs/reference/api/pandas.read_orc.html). 

Add those as supported filetypes [here](https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/develop/data_validation/clients.py#L117) and in our documentation.

### PR DESCRIPTION
**Pull request by luispavaogoogle**
Closes #1252 

/gcbrun

### CODE REVIEW

File: docs/connections.md [hunk 1]
```diff
@@ -242,7 +242,7 @@ data-validation connections add
     --database DATABASE                                 Redshift database
 ```
 
-## FileSystem (CSV or JSON only)
+## FileSystem (CSV, ORC, PARQUET or JSON only)
```
Comment (helensilva14) [inline docs/connections.md:245]: Besides the section title, line 253 also needs to be updated to add the new file types

File: tests/system/data_sources/test_filesystem.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
+# Copyright 2020 Google LLC
```
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
# Copyright 2024 Google LLC
```

File: tests/system/data_sources/test_filesystem.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+}
+
+
+def format_config_count(s_conn, t_conn):
+    config_count_valid = {
+        consts.CONFIG_SOURCE_CONN: s_conn,
+        consts.CONFIG_TARGET_CONN: t_conn,
+        # Validation Type
+        consts.CONFIG_TYPE: "Column",
+        # Configuration Required Depending on Validator Type
```
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
        # Configuration Required Depending on Validation Type
```

File: tests/system/data_sources/test_filesystem.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    """
+    schema_validation_test(tables="entries", tc="ORC_CONN")
+
+
+@mock.patch(
+    "data_validation.state_manager.StateManager.get_connection_config",
+    new=mock_get_connection_config,
+)
+def test_schema_validation_json():
+    """Test schema validation parquet file.
```
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
    """Test schema validation JSON file.
```

File: tests/system/data_sources/test_filesystem.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    """
+    schema_validation_test(tables="entries", tc="CSV_CONN")
+
+
+@mock.patch(
+    "data_validation.state_manager.StateManager.get_connection_config",
+    new=mock_get_connection_config,
+)
+def test_schema_validation_orc():
+    """Test schema validation parquet file.
```
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
    """Test schema validation ORC file.
```

File: tests/system/data_sources/test_filesystem.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+    """
+    schema_validation_test(tables="entries", tc="PARQUET_CONN")
+
+
+@mock.patch(
+    "data_validation.state_manager.StateManager.get_connection_config",
+    new=mock_get_connection_config,
+)
+def test_schema_validation_csv():
+    """Test schema validation parquet file.
```
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
    """Test schema validation CSV file.
```

File: tests/system/data_sources/test_filesystem.py [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+    assert df["source_agg_value"].equals(df["target_agg_value"])
+    assert sorted(list(df["source_agg_value"])) == ["28", "7", "7"]
+
+
+@mock.patch(
+    "data_validation.state_manager.StateManager.get_connection_config",
+    new=mock_get_connection_config,
+)
+def test_schema_validation_parquet():
+    """Test schema validation parquet file.
```
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
    """Test schema validation Parquet file.
```

File: tests/system/data_sources/test_filesystem.py [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+        verbose=False,
+    )
+    df = data_validator.execute()
+
+    assert df["source_agg_value"].equals(df["target_agg_value"])
+    assert sorted(list(df["source_agg_value"])) == ["28", "7", "7"]
+
+
+def test_filesystem_count_csv():
+
```
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
```

File: tests/system/data_sources/test_filesystem.py [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+        verbose=False,
+    )
+    df = data_validator.execute()
+
+    assert df["source_agg_value"].equals(df["target_agg_value"])
+    assert sorted(list(df["source_agg_value"])) == ["28", "7", "7"]
+
+
+def test_filesystem_count_orc():
+
```
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
```

File: tests/system/data_sources/test_filesystem.py [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+        verbose=False,
+    )
+    df = data_validator.execute()
+
+    assert df["source_agg_value"].equals(df["target_agg_value"])
+    assert sorted(list(df["source_agg_value"])) == ["28", "7", "7"]
+
+
+def test_filesystem_count_json():
+
```
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
```

File: tests/system/data_sources/test_filesystem.py [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+            },
+        ],
+        consts.CONFIG_FORMAT: "table",
+        consts.CONFIG_FILTER_STATUS: None,
+    }
+    return config_count_valid
+
+
+def test_filesystem_count_parquet():
+
```
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
```

**Review by helensilva14 — DISMISSED** [review]
Awesome work Luis! Just a few minor changes to do before I approve it

**Review by luispavaogoogle — COMMENTED** [review]
LGTM

File: tests/system/data_sources/test_filesystem.py [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+        return CSV_CONN
+    elif args[1] == "json-conn":
+        return JSON_CONN
+    elif args[1] == "orc-conn":
+        return ORC_CONN
+    else:
+        return PARQUET_CONN
+
+
+CSV_CONN = {
```
Comment (nj1973) [inline tests/system/data_sources/test_filesystem.py:41]: Nitpicky but constants should really be defined before and functions, i.e. these four constants should move above the `mock_get_connection_config` definition.

Approving anyway, I'll leave it up to you whether to move them or not.

**Review by nj1973 — APPROVED** [review]
LGTM aside from one nitpicky comment.

**Pull request merged**